src/main/java/org/opensearch/neuralsearch/processor/chunker/DelimiterChunker.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.chunker;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class DelimiterChunker implements IFieldChunker {
+
+    public DelimiterChunker() {}
+
+    public static String DELIMITER_FIELD = "delimiter";
+
+    public static String MAX_CHUNK_LIMIT_FIELD = "max_chunk_limit";
+
+    @Override
+    public void validateParameters(Map<String, Object> parameters) {
+        if (parameters.containsKey(DELIMITER_FIELD)) {
+            Object delimiter = parameters.get(DELIMITER_FIELD);
+            if (!(delimiter instanceof String)) {
+                throw new IllegalArgumentException("delimiter parameters: " + delimiter + " must be string.");
+            } else if (((String) delimiter).isEmpty()) {
+                throw new IllegalArgumentException("delimiter parameters should not be empty.");
+            }
+        } else {
+            throw new IllegalArgumentException("You must contain field: " + DELIMITER_FIELD + " in your parameter.");
+        }
+        if (parameters.containsKey(MAX_CHUNK_LIMIT_FIELD)) {
+            Object maxChunkLimit = parameters.get(MAX_CHUNK_LIMIT_FIELD);
+            if (!(maxChunkLimit instanceof Integer)) {
+                throw new IllegalArgumentException("Parameter max_chunk_limit:" + maxChunkLimit.toString() + " should be integer.");
+            } else if ((int) maxChunkLimit <= 0) {
+                throw new IllegalArgumentException("Parameter max_chunk_limit:" + maxChunkLimit + " is not greater than 0.");
```

Comment:
Since we are supporting nested map and list type input for the chunking processor, I am also planning to add max_chunk_limit parameter to the chunking processor.